app/src/citations.py

```diff
@@ -118,3 +120,37 @@ def remap_citation_ids(subsections: Sequence[Subsection], response: str) -> dict
         "\n  ".join([f"{id} -> {c.id}, {c.chunk.document.name}" for id, c in citations.items()]),
     )
     return citations
+
+
+def replace_citation_ids(response: str, remapped_citations: dict[str, Subsection]) -> str:
+    """Replace (citation-XX) in response with (citation-YY), where XX is the original citation ID
+    and YY is the remapped citation ID"""
+
+    def replace_citation(match: Match) -> str:
+        citation_id = match.group(1)
+        if citation_id not in remapped_citations:
+            logger.error(
+                "LLM generated a citation for a reference (%s) that doesn't exist.", citation_id
+            )
+            return ""
+        return "(citation-" + remapped_citations[citation_id].id + ")"
+
+    return re.sub(CITATION_PATTERN, replace_citation, response)
+
+
+@dataclass
+class ResponseWithSubsections:
+    response: str
+    subsections: Sequence[Subsection]
+
+
+def finalize_result(result: ResponseWithSubsections) -> ResponseWithSubsections:
```

Comment:
Could we give this a more descriptive name -- maybe `simplify_citation_numbers`?

Reply:
Will do